src/common/utility_mac.mm

```diff
@@ -255,4 +257,19 @@
 }
 #endif
 
+QString Utility::currentCpuArch()
+{
+    int ret = 0;
+    size_t size = sizeof(ret);
+
+    // ret will be 1 if the process is translated (most likely with rosetta2)
+    if (sysctlbyname("sysctl.proc_translated", &ret, &size, nullptr, 0) != -1) {
```

Comment:
No idea on how expansive this call is but as we add it to the user agent, could it make sense to cache the result locally in a static?
I don't expect this value to change at runtime :D

Reply:
Well, syscalls always cause a context switch, so yes, we should cache this.